dimod/constrained/constrained.py

```diff
@@ -1153,6 +1157,7 @@ def iter_constraint_data(self, sample_like: SamplesLike) -> Iterator[ConstraintD
         Args:
             sample_like: A sample. `sample-like` is an extension of
                 NumPy's array_like structure. See :func:`.as_samples`.
+            labels: A subset of the constraint labels over which to iterate.
```

Comment:
Might be worth mentioning that the constraints will be yielded in the same ordering as the labels passed, instead of in order of addition which I presume is the case without a labels subset request.

Reply:
I am being deliberately ambiguous in case we want to change the behavior later.